owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/drawer/DrawerViewModel.kt

```diff
@@ -62,13 +66,39 @@ class DrawerViewModel(
         return AccountUtils.getCurrentOwnCloudAccount(context)
     }
 
+    fun getUsernameOfAccount(accountName: String): String {
+        return AccountUtils.getUsernameOfAccount(accountName)
+    }
+
     fun setCurrentAccount(context: Context, accountName: String): Boolean {
         return AccountUtils.setCurrentOwnCloudAccount(context, accountName)
     }
 
-    fun deleteUnusedUserDirs(accounts: Array<Account>) {
+    fun removeAccount(context: Context) {
         viewModelScope.launch(coroutinesDispatcherProvider.io) {
-            FileStorageUtils.deleteUnusedUserDirs(accounts)
+            val loggedAccounts = AccountUtils.getAccounts(context)
+            FileStorageUtils.deleteUnusedUserDirs(loggedAccounts)
+
+            val userQuotas = getUserQuotasUseCase.execute(Unit)
+            val loggedAccountsNames = mutableListOf<String>()
+            val totalAccountsNames = mutableListOf<String>()
+            loggedAccounts.forEach { loggedAccount ->
+                loggedAccountsNames.add(loggedAccount.name)
+            }
+            userQuotas.forEach { account ->
+                totalAccountsNames.add(account.accountName)
+            }
```

Comment:
We could use `map`. What do you think?
```suggestion
            val loggedAccountsNames = loggedAccounts.map { it.name }
            val totalAccountsNames = userQuotas.map { it.accountName }
```